espflash/src/flasher/mod.rs

```diff
@@ -93,49 +107,89 @@ pub enum FlashMode {
 #[non_exhaustive]
 #[repr(u8)]
 #[strum(serialize_all = "SCREAMING_SNAKE_CASE")]
+#[doc(alias("esp_image_flash_size_t"))]
 pub enum FlashSize {
     /// 256 KB
-    _256Kb = 0x12,
+    _256Kb,
     /// 512 KB
-    _512Kb = 0x13,
+    _512Kb,
     /// 1 MB
-    _1Mb = 0x14,
+    _1Mb,
     /// 2 MB
-    _2Mb = 0x15,
+    _2Mb,
     /// 4 MB
     #[default]
-    _4Mb = 0x16,
+    _4Mb,
     /// 8 MB
-    _8Mb = 0x17,
+    _8Mb,
     /// 16 MB
-    _16Mb = 0x18,
+    _16Mb,
     /// 32 MB
-    _32Mb = 0x19,
+    _32Mb,
     /// 64 MB
-    _64Mb = 0x1a,
+    _64Mb,
     /// 128 MB
-    _128Mb = 0x21,
+    _128Mb,
+    /// 256 MB
+    _256Mb,
 }
 
 impl FlashSize {
-    fn from(value: u8) -> Result<FlashSize, Error> {
+    // components/esptool_py/esptool/docs/en/advanced-topics/firmware-image-format.rst:19
+    pub const fn encode_flash_size(self: FlashSize, chip: Chip) -> Result<u8, Error> {
+        use FlashSize::*;
+
+        let encoded = match chip {
+            Chip::Esp8266 => match self {
+                _256Kb => 1,
+                _512Kb => 0,
+                _1Mb => 2,
+                _2Mb => 3,
+                _4Mb => 4,
+                // Currently not supported
+                // _2Mb_c1 => 5,
+                // _4Mb_c1 => 6,
+                _8Mb => 8,
+                _16Mb => 9,
+                _ => return Err(Error::UnsupportedFlash(self as u8)),
+            },
+            _ => match self {
+                _1Mb => 0,
+                _2Mb => 1,
+                _4Mb => 2,
+                _8Mb => 3,
+                _16Mb => 4,
+                _32Mb => 5,
+                _64Mb => 6,
+                _128Mb => 7,
+                _256Mb => 8,
+                _ => return Err(Error::UnsupportedFlash(self as u8)),
+            },
+        };
+
+        Ok(encoded)
+    }
+
+    // components/esptool_py/esptool/esptool.py:151
```

Comment:
Do you mind sharing a link to this reference? I assume this belongs to the esp-idf repo, but in `master` there is no such line in the esptool.py

Reply:
Found it, here is a link to where it now lives: https://github.com/espressif/esptool/blob/f4d2510e2c897621884f433ef3f191e8fc5ff184/esptool/cmds.py#L42

Reply:
On the one hand, permanent link is better because of the line, on the other hand, it would be nice to have a link to master version. I guess approach with the permalink is more durable